go/libraries/doltcore/sqle/binlogreplication/binlog_replica_applier.go

```diff
@@ -254,65 +258,68 @@ func (a *binlogReplicaApplier) replicaBinlogEventHandler(ctx *sql.Context) error
 	}
 	engine := server.Engine
 
-	conn, err := a.connectAndStartReplicationEventStream(ctx)
-	if err != nil {
-		return err
-	}
+	var conn *mysql.Conn
+	var eventProducer *binlogEventProducer
+	var binlogEventChan = make(chan mysql.BinlogEvent)
```

Comment:
We never close these channels, which could leak resources over time (not sure if open channels get cleaned up when they go out of scope or not)

Best practice is for the send-side of a channel to manage its lifecycle including closing it. One reasonable way to do this would be for the Go method of eventProducer to return both these methods, and to close them when it is stopped.